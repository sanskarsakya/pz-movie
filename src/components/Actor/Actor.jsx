import React from "react";

function Actor({ name, image, character }) {
  return (
    <div data-testid="actor" className="w-1/5 p-3 transform hover:-translate-y-2 ease-in-out duration-200 transition-translate">
      <div
        className="rounded-md overflow-hidden"
        style={{ height: 200, borderRadius: 20 }}
      // width:270
      >
        <img
          className="object-cover"
          src={"https://image.tmdb.org/t/p/w185" + image}
          alt=""
        />
      </div>
      <p data-testid="actor_name" className="mt-4 text-xl font-semibold cursor-pointer">{name}</p>
      <div
        className="mt-1 text-sm tracking-wider flex items-center"
        style={{ color: "rgba(255, 255, 255, 0.8)" }}
      >
        <p>{character}</p>
      </div>
    </div>
  );
};

export default Actor;
