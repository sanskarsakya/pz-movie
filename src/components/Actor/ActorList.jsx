import React from "react";
import Actor from "./Actor";

export const ActorList = ({ actors }) => {
  return (
    <>
      {/* start cast and crew */}
      <div className="mt-12">
        <h1 className="font-semibold text-3xl tracking-wider mr-2">
          Cast and crew
        </h1>
        {actors && (
          <div className="-ml-3 flex justify-between flex-wrap mt-5">
            {actors.map((actor, index) => (
              <Actor
                key={actor.id}
                name={actor.name}
                character={actor.character}
                image={actor.profile_path}
              />
            ))}
          </div>
        )}
      </div>
      {/* end cast and crew */}
    </>
  );
};
