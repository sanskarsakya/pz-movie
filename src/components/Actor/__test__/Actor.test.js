import React from "react";
import ReactDOM from "react-dom";

import Actor from "../Actor";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

afterEach(cleanup);


describe("Actor test suite", () => {

    it("render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Actor />, div);
    });

    it("renders actor correctly", () => {
        const { getByTestId } = render(<Actor name="puzan sakya" />);
        expect(getByTestId("actor_name")).toHaveTextContent("puzan sakya");
    });

    it("renders actor correctly", () => {
        const { getByTestId } = render(<Actor name="test name" />);
        expect(getByTestId("actor_name")).toHaveTextContent("test name");
    });

    it("match actor snapshot", () => {
        const tree = renderer.create(<Actor name="test" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
