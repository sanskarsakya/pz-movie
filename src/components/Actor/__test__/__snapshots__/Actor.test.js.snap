// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actor test suite match actor snapshot 1`] = `
<div
  className="w-1/5 p-3 transform hover:-translate-y-2 ease-in-out duration-200 transition-translate"
  data-testid="actor"
>
  <div
    className="rounded-md overflow-hidden"
    style={
      Object {
        "borderRadius": 20,
        "height": 200,
      }
    }
  >
    <img
      alt=""
      className="object-cover"
      src="https://image.tmdb.org/t/p/w185undefined"
    />
  </div>
  <p
    className="mt-4 text-xl font-semibold cursor-pointer"
    data-testid="actor_name"
  >
    test
  </p>
  <div
    className="mt-1 text-sm tracking-wider flex items-center"
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.8)",
      }
    }
  >
    <p />
  </div>
</div>
`;
